File: Java/ContarCaracteres/src/ContarCaracteres.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ContarCaracteres {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingresa un texto: ");
        String texto = scanner.nextLine();

        // Contar la cantidad total de caracteres
        int totalCaracteres = texto.length();

        // Contar la cantidad de signos ortográficos (no letras ni números)
        int signosOrtograficos = contarSignosOrtograficos(texto);

        // Crear un mapa para almacenar la frecuencia de cada letra
        Map<Character, Integer> frecuenciaLetras = new HashMap<>();

        // Iterar a través del texto para contar las letras
        for (char caracter : texto.toCharArray()) {
            if (Character.isLetter(caracter)) {
                char letra = Character.toLowerCase(caracter); // Tratar las letras como minúsculas para evitar distinción entre mayúsculas y minúsculas
                frecuenciaLetras.put(letra, frecuenciaLetras.getOrDefault(letra, 0) + 1);
            }
        }

        // Mostrar resultados
        System.out.println("Cantidad total de caracteres: " + totalCaracteres);
        System.out.println("Cantidad de signos ortográficos: " + signosOrtograficos);

        System.out.println("Frecuencia de letras (en orden alfabético):");
        frecuenciaLetras.entrySet().stream()
                .sorted(Map.Entry.comparingByKey())
                .forEach(entry -> System.out.println(entry.getKey() + ": " + entry.getValue()));
    }

    static int contarSignosOrtograficos(String texto) {
        // Utiliza una expresión regular para contar signos ortográficos
        Pattern pattern = Pattern.compile("\\P{Alnum}");
        Matcher matcher = pattern.matcher(texto);
        int count = 0;
        while (matcher.find()) {
            count++;
        }
        return count;
    }
}

File: Python/ContarCaracteres/ContarCaracteres.py
# Pedir al usuario que ingrese un texto
texto = input("Ingresa un texto: ")

# Contar la cantidad total de caracteres
cantidad_total_caracteres = len(texto)

# Crear un diccionario para contar la cantidad de cada letra
conteo_letras = {}

# Recorrer el texto y contar las letras
for letra in texto:
    if letra.isalpha():  # Verificar si el carácter es una letra
        letra = letra.lower()  # Convertir la letra a minúscula para contar sin distinción de mayúsculas/minúsculas
        if letra in conteo_letras:
            conteo_letras[letra] += 1
        else:
            conteo_letras[letra] = 1

# Mostrar la cantidad total de caracteres
print(f"La cantidad total de caracteres en el texto es: {cantidad_total_caracteres}")

# Mostrar la cantidad de cada letra
print("Cantidad de cada letra:")
for letra, cantidad in conteo_letras.items():
    print(f"{letra}: {cantidad}")
